File: py/hanju7.py
# -*- coding: utf-8 -*-
import json
import re
import requests
from bs4 import BeautifulSoup

class Spider(object):
    """
    新韩剧网(hanju7.com)爬虫 Python版
    源JS作者：deepseek
    转换：AI Assistant
    """
    
    def __init__(self):
        self.siteUrl = 'https://www.hanju7.com'
        self.headers = {
            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36',
            'Referer': self.siteUrl
        }

    def init(self, extend=""):
        pass

    def homeContent(self, filter):
        result = {}
        cateManual = {
            "韩剧": "1",
            "韩国电影": "3",
            "韩国综艺": "4",
            "排行榜": "hot",
            "最新更新": "new"
        }
        classes = []
        for k, v in cateManual.items():
            classes.append({
                'type_name': k,
                'type_id': v
            })

        # 筛选配置
        years = [
            {"n": "全部", "v": ""}, {"n": "2025", "v": "2025"}, {"n": "2024", "v": "2024"},
            {"n": "2023", "v": "2023"}, {"n": "2022", "v": "2022"}, {"n": "2021", "v": "2021"},
            {"n": "2020", "v": "2020"}, {"n": "10后", "v": "2010__2019"}, {"n": "00后", "v": "2000__2009"},
            {"n": "90后", "v": "1990__1999"}, {"n": "80后", "v": "1980__1989"}, {"n": "更早", "v": "1900__1980"}
        ]
        sorts = [{"n": "最新", "v": "newstime"}, {"n": "热门", "v": "onclick"}]

        filterConfig = {}
        # 为 ID 1, 3, 4 添加筛选
        for type_id in ["1", "3", "4"]:
            filterConfig[type_id] = [
                {"key": "year", "name": "按年份", "value": years},
                {"key": "sort", "name": "按排序", "value": sorts}
            ]

        result['class'] = classes
        result['filters'] = filterConfig
        return result

    def homeVideoContent(self):
        try:
            r = requests.get(self.siteUrl + '/', headers=self.headers)
            r.encoding = 'utf-8'
            soup = BeautifulSoup(r.text, 'html.parser')
            videos = self._parse_video_list(soup)
            return {'list': videos}
        except Exception as e:
            return {'list': []}

    def categoryContent(self, tid, pg, filter, extend):
        result = {}
        # 获取筛选参数
        year = extend.get('year', '')
        sort = extend.get('sort', '')

        try:
            url = ""
            if tid in ['hot', 'new']:
                # 排行榜或最新更新
                url = f"{self.siteUrl}/{tid}.html"
                r = requests.get(url, headers=self.headers)
                r.encoding = 'utf-8'
                soup = BeautifulSoup(r.text, 'html.parser')
                
                videos = []
                # JS 逻辑：querySelectorAll('#t ~ li')
                # 这里的逻辑需要根据HTML结构适配，通常是一个列表容器
                # 假设HTML结构中有一个 id="t" 的元素，后面紧跟 li
                # Python BS4 处理兄弟节点不如 JS 方便，通常直接找包含列表的 ul 或 div
                
                # 尝试查找主要的列表容器，通常是 .list 或 similar
                # 兼容 JS 中的逻辑，尝试找到列表项
                items = soup.select('.list li') # 假设通用类名，如果不匹配需根据实际网页调整
                if not items:
                    # 如果没有 .list，尝试直接找 #t 后的兄弟节点（模拟JS逻辑）
                    t_elem = soup.find(id='t')
                    if t_elem:
                        items = t_elem.find_next_siblings('li')

                i = 1
                for li in items:
                    a_tag = li.find('a', href=re.compile(r'/detail/'))
                    if not a_tag:
                        continue
                        
                    vid = self.siteUrl + a_tag['href']
                    name = a_tag.get_text(strip=True)
                    
                    time_elem = li.find(id='time')
                    actor_elem = li.find(id='actor')
                    
                    year_txt = time_elem.get_text(strip=True) if time_elem else ""
                    actor_txt = actor_elem.get_text(strip=True) if actor_elem else ""
                    
                    # 模拟JS中的占位图逻辑
                    pic = f'https://dummyimage.com/240x240/ffffff/ffc800.png&text={i}'
                    i += 1
                    
                    videos.append({
                        "vod_id": vid,
                        "vod_name": name,
                        "vod_year": year_txt,
                        "vod_pic": pic,
                        "vod_remarks": f"{year_txt} | {actor_txt}"
                    })
                
                result = {
                    "code": 1, 
                    "msg": "数据列表", 
                    "list": videos, 
                    "page": 1, 
                    "pagecount": 1, 
                    "limit": len(videos), 
                    "total": len(videos)
                }

            else:
                # 普通分类
                # URL格式: /list/tid-year-sort-pg.html
                url = f"{self.siteUrl}/list/{tid}-{year}-{sort}-{pg}.html"
                r = requests.get(url, headers=self.headers)
                r.encoding = 'utf-8'
                soup = BeautifulSoup(r.text, 'html.parser')
                videos = self._parse_video_list(soup)
                
                result = {
                    "code": 1, 
                    "msg": "数据列表", 
                    "list": videos, 
                    "page": int(pg), 
                    "pagecount": 10000, 
                    "limit": 24, 
                    "total": 240000
                }
                
        except Exception as e:
            print(f"Error in categoryContent: {e}")
            result = {"list": []}
            
        return result

    def detailContent(self, array):
        try:
            tid = array[0]
            if not tid.startswith('http'):
                tid = self.siteUrl + tid if tid.startswith('/') else tid
                
            r = requests.get(tid, headers=self.headers)
            r.encoding = 'utf-8'
            soup = BeautifulSoup(r.text, 'html.parser')

            # 获取文本辅助函数
            def get_dt_text(keyword):
                dt = soup.find('dt', string=lambda t: t and keyword in t)
                if dt and dt.find_next_sibling():
                    return dt.find_next_sibling().get_text(strip=True)
                return ""

            # 提取图片
            img_tag = soup.select_one('img[src*="//pics.hanju7.com/"]')
            pic = 'https:' + img_tag['src'] if img_tag else ''

            # 提取剧情
            content_div = soup.select_one('.juqing')
            content = content_div.get_text(strip=True) if content_div else ""

            # 提取播放列表 (Regex 解析 bb_a 函数)
            # JS: bb_a('链接','名称') -> 实际上原JS正则是 bb_a('name','url')
            # JS Match: match[1] = url/param, match[2] = name (Wait, let's re-read JS)
            # JS Source: match(/bb_a\('([^']+)','([^']+)'/) -> match[1] is 1st arg, match[2] is 2nd arg.
            # JS Logic: `${match[2]}$${match[1]}`
            # 这里的顺序取决于网页源码中 bb_a 的参数顺序。通常是 bb_a('集数名称', '播放地址')
            
            play_urls = []
            scripts = soup.find_all('script') # 或者直接在html文本中搜索
            # 在整个HTML中查找所有 onclick="bb_a(...)"
            pattern = re.compile(r"bb_a\('([^']+?)','([^']+?)'\)")
            # 查找所有匹配项
            matches = pattern.findall(r.text)
            
            # matches list of tuples: [(arg1, arg2), ...]
            # 假设网页是 bb_a('第01集', '/play/xxx.html')
            # 那么 arg1=名称, arg2=URL
            # JS return: match[2]$match[1] => URL$名称
            
            for m in matches:
                name = m[0]
                url_part = m[1]
                play_urls.append(f"{url_part}${name}")

            vod = {
                "vod_id": tid,
                "vod_name": get_dt_text('片名'),
                "vod_pic": pic,
                "vod_remarks": get_dt_text('状态'),
                "vod_year": get_dt_text('上映').split('-')[0],
                "vod_actor": get_dt_text('主演'),
                "vod_content": content,
                "vod_play_from": "新韩剧网",
                "vod_play_url": "#".join(play_urls)
            }

            return {
                "code": 1, 
                "msg": "数据列表", 
                "page": 1, 
                "pagecount": 1, 
                "limit": 1, 
                "total": 1, 
                "list": [vod]
            }
            
        except Exception as e:
            print(f"Error in detailContent: {e}")
            return {"list": []}

    def searchContent(self, key, quick, pg="1"):
        try:
            url = f"{self.siteUrl}/search/"
            data = {
                "show": "searchkey",
                "keyboard": key
            }
            r = requests.post(url, data=data, headers=self.headers)
            r.encoding = 'utf-8'
            soup = BeautifulSoup(r.text, 'html.parser')

            videos = []
            # 搜索结果列表解析
            # JS: querySelectorAll('#t ~ li')
            # 兼容处理，寻找列表
            items = soup.select('.list li') 
            if not items:
                # 尝试寻找搜索结果容器
                t_elem = soup.find(id='t')
                if t_elem:
                    items = t_elem.find_next_siblings('li')

            for li in items:
                a_tag = li.find('a', href=re.compile(r'/detail/'))
                if not a_tag:
                    continue
                
                vid = self.siteUrl + a_tag['href']
                name = a_tag.get('title', a_tag.get_text(strip=True))
                
                time_elem = li.find(id='time')
                actor_elem = li.find(id='actor')
                
                year_txt = time_elem.get_text(strip=True) if time_elem else ""
                
                # 提取年份正则
                year_match = re.search(r'\((\d{4})\)', a_tag.get_text())
                year_in_title = year_match.group(1) if year_match else ""
                
                actor_txt = actor_elem.get_text(strip=True) if actor_elem else ""
                remarks = f"{year_in_title} | {actor_txt}"

                videos.append({
                    "vod_id": vid,
                    "vod_name": name,
                    "vod_year": year_txt,
                    "vod_remarks": remarks,
                    # 搜索列表可能没有图片，或者需要额外解析，这里JS原版也没处理图片
                    "vod_pic": "" 
                })

            return {
                "code": 1, 
                "msg": "搜索结果", 
                "list": videos, 
                "page": 1, 
                "pagecount": 1, 
                "limit": len(videos), 
                "total": len(videos)
            }
        except Exception as e:
            print(f"Error in searchContent: {e}")
            return {"list": []}

    def playerContent(self, flag, id, vipFlags):
        # id 是 vod_play_url 中的 url 部分
        # 这里的 id 应该是 /play/xxxxx.html 这样的格式
        
        url = id
        if not url.startswith('http'):
            url = self.siteUrl + url

        # JS 版本使用了 webview sniffing (type: 'sniff')
        # 并配合了脚本点击。
        # 在 Python 爬虫标准中，我们返回 parse=1 让播放器/壳子去嗅探目标网页中的视频流
        
        return {
            "parse": 1,
            "playUrl": "",
            "url": url,
            "header": {
                "User-Agent": self.headers['User-Agent'],
                "Referer": self.siteUrl
            }
        }

    def _parse_video_list(self, soup):
        videos = []
        items = soup.select('.list li')
        for li in items:
            a_tag = li.find('a', href=re.compile(r'/detail/'))
            if not a_tag:
                continue
            
            vid = self.siteUrl + a_tag['href']
            name = a_tag.get('title', '')
            if not name:
                name = a_tag.get_text(strip=True)
                
            pic_url = a_tag.get('data-original', '')
            if pic_url and not pic_url.startswith('http'):
                pic_url = 'https:' + pic_url
                
            tip_elem = li.select_one('.tip')
            year = tip_elem.get_text(strip=True) if tip_elem else ""
            
            videos.append({
                "vod_id": vid,
                "vod_name": name,
                "vod_pic": pic_url,
                "vod_year": year
            })
        return videos

File: js/hanju7.js
/**
 * 韩剧网(hanju7.com)爬虫
 * 作者：基于ddys爬虫模板适配
 * 版本：1.0
 * 最后更新：2025-12-17
 */

function hanju7Spider() {
    const baseUrl = 'https://www.hanju7.com';
    
    // 延时函数
    const delay = (ms) => new Promise(resolve => setTimeout(resolve, ms));
    
    /**
     * 提取视频列表数据（私有方法）
     * @param {Document} document - DOM文档对象
     * @param {string} selector - 选择器，默认为'.list li'
     * @returns {Array} 视频数据数组
     */
    const extractVideos = (document, selector = '.list li') => {
        return Array.from(document.querySelectorAll(selector)).map(li => {
            const linkEl = li.querySelector('a.tu');
            const titleEl = li.querySelector('p a');
            const tipEl = li.querySelector('.tip');
            const actorEl = li.querySelector('p:nth-child(3)');
            
            // 提取图片URL
            let vod_pic = '';
            if (linkEl) {
                vod_pic = linkEl.getAttribute('data-original') || 
                          linkEl.style.backgroundImage?.match(/url\(["']?([^"')]+)["']?\)/)?.[1] || '';
            }
            
            // 提取详情页链接
            let vod_id = '';
            if (linkEl && linkEl.href) {
                vod_id = linkEl.href.startsWith('http') ? linkEl.href : baseUrl + (linkEl.href.startsWith('/') ? '' : '/') + linkEl.href;
            } else if (titleEl && titleEl.href) {
                vod_id = titleEl.href.startsWith('http') ? titleEl.href : baseUrl + (titleEl.href.startsWith('/') ? '' : '/') + titleEl.href;
            }
            
            // 提取标题
            let vod_name = '';
            if (titleEl) {
                vod_name = titleEl.title || titleEl.textContent || '';
            } else if (linkEl) {
                vod_name = linkEl.title || '';
            }
            
            // 生成视频ID（用于详情页）
            let videoId = '';
            if (vod_id) {
                const match = vod_id.match(/\/detail\/(\d+)\.html/);
                if (match) {
                    videoId = match[1];
                }
            }
            
            return {
                vod_id: videoId || vod_name.replace(/[^\w]/g, '_'),
                vod_name: vod_name,
                vod_pic: vod_pic,
                vod_remarks: tipEl?.textContent?.trim() || '',
                vod_actor: actorEl?.textContent?.trim() || '',
                vod_href: vod_id // 保存完整链接用于详情页
            };
        }).filter(video => video.vod_name && video.vod_pic); // 过滤无效数据
    };
    
    /**
     * 提取排行榜数据
     * @param {Document} document - DOM文档对象
     * @returns {Array} 排行榜视频数据
     */
    const extractRankVideos = (document) => {
        const rankVideos = [];
        
        // 提取韩剧榜
        const kdramaRankList = document.querySelectorAll('.box:nth-child(1) .list_txt li');
        kdramaRankList.forEach(li => {
            const spanEl = li.querySelector('span');
            const aEl = li.querySelector('a');
            const iEl = li.querySelector('i');
            
            if (aEl && aEl.href) {
                const vod_id = aEl.href.match(/\/detail\/(\d+)\.html/)?.[1] || '';
                rankVideos.push({
                    vod_id: vod_id,
                    vod_name: aEl.textContent.replace(/^\d+\s*/, '').trim(),
                    vod_remarks: spanEl?.textContent?.trim() || '',
                    vod_pic: `//pics.hanju7.com/pics/${vod_id}.jpg`,
                    vod_href: baseUrl + (aEl.href.startsWith('/') ? '' : '/') + aEl.href,
                    vod_type: '韩剧',
                    vod_rank: iEl?.textContent?.trim() || ''
                });
            }
        });
        
        // 提取韩影榜
        const movieRankList = document.querySelectorAll('.box:nth-child(2) .list_txt li');
        movieRankList.forEach(li => {
            const spanEl = li.querySelector('span');
            const aEl = li.querySelector('a');
            const iEl = li.querySelector('i');
            
            if (aEl && aEl.href) {
                const vod_id = aEl.href.match(/\/detail\/(\d+)\.html/)?.[1] || '';
                rankVideos.push({
                    vod_id: vod_id,
                    vod_name: aEl.textContent.replace(/^\d+\s*/, '').trim(),
                    vod_remarks: spanEl?.textContent?.trim() || '',
                    vod_pic: `//pics.hanju7.com/pics/${vod_id}.jpg`,
                    vod_href: baseUrl + (aEl.href.startsWith('/') ? '' : '/') + aEl.href,
                    vod_type: '韩影',
                    vod_rank: iEl?.textContent?.trim() || ''
                });
            }
        });
        
        // 提取韩综榜
        const varietyRankList = document.querySelectorAll('.box:nth-child(3) .list_txt li');
        varietyRankList.forEach(li => {
            const spanEl = li.querySelector('span');
            const aEl = li.querySelector('a');
            const iEl = li.querySelector('i');
            
            if (aEl && aEl.href) {
                const vod_id = aEl.href.match(/\/detail\/(\d+)\.html/)?.[1] || '';
                rankVideos.push({
                    vod_id: vod_id,
                    vod_name: aEl.textContent.replace(/^\d+\s*/, '').trim(),
                    vod_remarks: spanEl?.textContent?.trim() || '',
                    vod_pic: `//pics.hanju7.com/pics/${vod_id}.jpg`,
                    vod_href: baseUrl + (aEl.href.startsWith('/') ? '' : '/') + aEl.href,
                    vod_type: '韩综',
                    vod_rank: iEl?.textContent?.trim() || ''
                });
            }
        });
        
        return rankVideos;
    };

    return {
        async init(cfg) {
            return {
                webview: {
                    debug: true,
                    showWebView: false,
                    widthPercent: 80,
                    heightPercent: 60,
                    keyword: '',
                    returnType: 'dom',
                    timeout: 30,
                    blockImages: true,
                    enableJavaScript: true,
                    header: {
                        'Referer': baseUrl,
                        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
                    }
                }
            };
        },
        
        async homeContent(filter) {
            console.log("homeContent 开始执行, filter:", filter);

            return {
                class: [
                    { type_id: "1", type_name: "韩剧" },
                    { type_id: "3", type_name: "韩国电影" },
                    { type_id: "4", type_name: "韩国综艺" },
                    { type_id: "hot", type_name: "排行榜" },
                    { type_id: "new", type_name: "最近更新" }
                ]
            };
        },
        
        async homeVideoContent() {
            console.log("homeVideoContent 开始执行");

            try {
                const document = await Java.wvOpen(baseUrl + '/');
                
                // 提取首页的三个板块
                const allVideos = [];
                
                // 韩剧板块
                const kdramaSection = document.querySelector('.box:nth-child(1) .list');
                if (kdramaSection) {
                    const kdramaVideos = extractVideos(kdramaSection);
                    allVideos.push(...kdramaVideos.map(v => ({...v, vod_type: '韩剧'})));
                }
                
                // 韩影板块
                const movieSection = document.querySelector('.box:nth-child(2) .list');
                if (movieSection) {
                    const movieVideos = extractVideos(movieSection);
                    allVideos.push(...movieVideos.map(v => ({...v, vod_type: '韩影'})));
                }
                
                // 韩综板块
                const varietySection = document.querySelector('.box:nth-child(3) .list');
                if (varietySection) {
                    const varietyVideos = extractVideos(varietySection);
                    allVideos.push(...varietyVideos.map(v => ({...v, vod_type: '韩综'})));
                }
                
                // 提取排行榜数据
                const rankVideos = extractRankVideos(document);
                allVideos.push(...rankVideos);
                
                console.log("首页取到的视频列表，共", allVideos.length, "条");
                return { list: allVideos };
            } catch (error) {
                console.error("homeVideoContent 出错:", error);
                return { list: [] };
            }
        },
        
        async categoryContent(tid, pg, filter, extend) {
            console.log(`categoryContent - tid:${tid}, pg:${pg}`);
            
            try {
                let url = '';
                if (tid === 'hot') {
                    url = `${baseUrl}/hot.html`;
                } else if (tid === 'new') {
                    url = `${baseUrl}/new.html`;
                } else {
                    url = `${baseUrl}/list/${tid}---${pg}---.html`;
                }
                
                const document = await Java.wvOpen(url);
                
                // 提取列表数据
                const videos = extractVideos(document);
                
                // 提取分页信息
                let page = pg || 1;
                let pagecount = 1;
                let total = videos.length;
                
                // 尝试获取分页信息
                const pageElement = document.querySelector('.pages a:last-child');
                if (pageElement) {
                    const pageText = pageElement.textContent;
                    const pageMatch = pageText.match(/\d+/);
                    if (pageMatch) {
                        pagecount = parseInt(pageMatch[0]);
                    }
                }
                
                console.log(`categoryContent - 获取到 ${videos.length} 条数据`);
                
                return {
                    code: 1,
                    msg: "数据列表",
                    list: videos,
                    page: parseInt(page),
                    pagecount: pagecount,
                    limit: 20,
                    total: total
                };
            } catch (error) {
                console.error("categoryContent 出错:", error);
                return { code: 0, msg: "获取数据失败", list: [] };
            }
        },
        
        async detailContent(ids) {
            console.log("detailContent - ids:", ids);
            
            try {
                const id = ids[0];
                let url = '';
                
                // 判断是完整URL还是ID
                if (id.startsWith('http')) {
                    url = id;
                } else if (id.match(/^\d+$/)) {
                    url = `${baseUrl}/detail/${id}.html`;
                } else {
                    // 尝试从vod_href中获取
                    url = id;
                }
                
                const document = await Java.wvOpen(url);
                
                // 提取基本信息
                const title = document.querySelector('.detail h1')?.textContent?.trim() || 
                             document.querySelector('.detail .title')?.textContent?.trim() || '';
                
                // 提取图片
                const imgEl = document.querySelector('.pic img, .detail-pic img');
                let vod_pic = '';
                if (imgEl) {
                    vod_pic = imgEl.getAttribute('src') || 
                             imgEl.getAttribute('data-src') || 
                             imgEl.getAttribute('data-original') || '';
                }
                
                if (!vod_pic && id.match(/^\d+$/)) {
                    vod_pic = `//pics.hanju7.com/pics/${id}.jpg`;
                }
                
                // 提取详细信息
                let vod_area = '', vod_year = '', vod_actor = '', vod_director = '', 
                    vod_remarks = '', vod_lang = '', vod_content = '';
                
                const infoElements = document.querySelectorAll('.info p, .detail-info p, .info li');
                infoElements.forEach(el => {
                    const text = el.textContent.trim();
                    if (text.includes('地区：')) vod_area = text.replace('地区：', '').trim();
                    else if (text.includes('年份：')) vod_year = text.replace('年份：', '').trim();
                    else if (text.includes('主演：')) vod_actor = text.replace('主演：', '').trim();
                    else if (text.includes('导演：')) vod_director = text.replace('导演：', '').trim();
                    else if (text.includes('语言：')) vod_lang = text.replace('语言：', '').trim();
                    else if (text.includes('状态：')) vod_remarks = text.replace('状态：', '').trim();
                    else if (text.includes('更新：')) vod_remarks = text.replace('更新：', '').trim();
                });
                
                // 提取简介
                const contentEl = document.querySelector('.content, .intro, .detail-content');
                if (contentEl) {
                    vod_content = contentEl.textContent.trim();
                }
                
                // 提取播放列表
                const playlists = [];
                const playElements = document.querySelectorAll('.playlist, .play-list, .downlist');
                
                playElements.forEach((playlist, index) => {
                    const episodes = [];
                    const links = playlist.querySelectorAll('a');
                    
                    links.forEach(a => {
                        const episodeName = a.textContent.trim();
                        let episodeUrl = a.getAttribute('href') || '';
                        
                        if (episodeUrl && !episodeUrl.startsWith('http')) {
                            episodeUrl = baseUrl + (episodeUrl.startsWith('/') ? '' : '/') + episodeUrl;
                        }
                        
                        if (episodeName && episodeUrl) {
                            episodes.push(`${episodeName}$${episodeUrl}`);
                        }
                    });
                    
                    if (episodes.length > 0) {
                        playlists.push({
                            title: `线路${index + 1}`,
                            episodes: episodes
                        });
                    }
                });
                
                // 如果没有找到播放列表，尝试其他选择器
                if (playlists.length === 0) {
                    const allLinks = document.querySelectorAll('a[href*="play"]');
                    allLinks.forEach(a => {
                        if (a.textContent.includes('播放') || a.href.includes('play')) {
                            playlists.push({
                                title: '在线播放',
                                episodes: [`播放$${a.href}`]
                            });
                        }
                    });
                }
                
                // 构建视频对象
                const vod = {
                    vod_id: id.match(/^\d+$/) ? id : id.replace(/[^\w]/g, '_'),
                    vod_name: title,
                    vod_pic: vod_pic,
                    vod_remarks: vod_remarks,
                    vod_year: vod_year,
                    vod_actor: vod_actor,
                    vod_director: vod_director,
                    vod_area: vod_area,
                    vod_lang: vod_lang || '韩语',
                    vod_content: vod_content,
                    vod_play_from: playlists.map(p => p.title).join('$$$'),
                    vod_play_url: playlists.map(p => p.episodes.join('#')).join('$$$'),
                    type_name: '' // 将在下面判断
                };
                
                // 判断类型
                if (url.includes('/detail/')) {
                    const typeMatch = url.match(/\/list\/(\d+)---/);
                    if (typeMatch) {
                        const typeId = typeMatch[1];
                        if (typeId === '1') vod.type_name = '韩剧';
                        else if (typeId === '3') vod.type_name = '韩国电影';
                        else if (typeId === '4') vod.type_name = '韩国综艺';
                    }
                }
                
                return {
                    code: 1,
                    msg: "数据列表",
                    page: 1,
                    pagecount: 1,
                    limit: 1,
                    total: 1,
                    list: [vod]
                };
                
            } catch (error) {
                console.error("detailContent 出错:", error);
                return {
                    code: 0,
                    msg: "获取详情失败",
                    list: []
                };
            }
        },
        
        async searchContent(key, quick, pg) {
            console.log("searchContent - key:", key, "pg:", pg);
            
            try {
                const encodedKey = encodeURIComponent(key);
                const url = `${baseUrl}/search/?wd=${encodedKey}&page=${pg || 1}`;
                
                const document = await Java.wvOpen(url);
                
                // 提取搜索结果
                const videos = extractVideos(document, '.search-list li');
                
                // 如果没有找到，尝试其他选择器
                if (videos.length === 0) {
                    const fallbackVideos = extractVideos(document, '.list li');
                    return {
                        code: 1,
                        msg: "搜索结果",
                        list: fallbackVideos,
                        page: pg || 1,
                        pagecount: 1,
                        limit: 20,
                        total: fallbackVideos.length
                    };
                }
                
                return {
                    code: 1,
                    msg: "搜索结果",
                    list: videos,
                    page: pg || 1,
                    pagecount: 1,
                    limit: 20,
                    total: videos.length
                };
            } catch (error) {
                console.error("searchContent 出错:", error);
                return { code: 0, msg: "搜索失败", list: [] };
            }
        },
        
        async playerContent(flag, id, vipFlags) {
            console.log("playerContent - flag:", flag, "id:", id);
            
            // 韩剧网直接返回原始播放链接
            return { 
                url: id, 
                parse: 1,  // 表示需要解析
                header: {
                    'Referer': baseUrl,
                    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
                }
            };
        },
        
        async action(actionStr) {
            console.log("action - actionStr:", actionStr);
            try {
                const params = JSON.parse(actionStr);
                console.log("action params:", params);
                // 这里可以根据action参数执行特定操作
                return { code: 1, msg: "操作成功", list: [] };
            } catch (e) {
                console.log("action is not JSON, treat as string");
                return { code: 0, msg: "操作失败", list: [] };
            }
        }
    };
}

const spider = hanju7Spider();
spider;
